### PR TITLE
Closes #20 – Remove Cart Item

### LINKED ISSUE #20: Bug: Can't remove an item from the cart
Customers are complaining that they can't remove an item from the cart after it has been added.

When the client performs a DELETE request to the `/lineitems/n` resource, it should remove the specified item from the user's cart.

### PR DESCRIPTION
**Pull request by gqgonzales**
## Changes

- In `bangazonapi/views/lineitem.py `, adjusted depth of response from postman, added list method, and delete logic

## Requests / Responses

**Request**

GET `http://localhost:8000/lineitems/4` to confirm you can return a single line item from `order_id: 2`

**Response**

HTTP/1.1 201 OK

```json
{
    "id": 4,
    "url": "http://localhost:8000/lineitems/4",
    "order": {
        "url": "http://localhost:8000/orders/2",
        "created_date": "2019-04-12",
        "customer": {
            "url": "http://localhost:8000/customers/7",
            "phone_number": "555-1212",
            "address": "100 Indefatiguable Way",
            "user": "http://localhost:8000/users/8"
        },
        "payment_type": null
    },
    "product": {
        "url": "http://localhost:8000/products/52",
        "deleted": null,
        "name": "900",
        "price": 1296.98,
        "description": "1987 Saab",
        "quantity": 2,
        "created_date": "2019-03-19",
        "location": "Vratsa",
        "image_path": null,
        "customer": {
            "url": "http://localhost:8000/customers/7",
            "phone_number": "555-1212",
            "address": "100 Indefatiguable Way",
            "user": "http://localhost:8000/users/8"
        },
        "category": {
            "url": "http://localhost:8000/productcategories/2",
            "name": "Auto"
        }
    }
}
```

------------------------------------------

**Request**

DELETE `http://localhost:8000/lineitems/4` to destroy the order product instance

**Response**

HTTP/1.1 204 NO CONTENT

```
{}
```

## Testing

Steps to Test:

- [ ] Run migrations
- [ ] Seed database with `python manage.py loaddata product order_product order`
- [ ] Run server with `python manage.py runserver`
- [ ] Run GET on `http://localhost:8000/profile/cart`; in Headers, pass: KEY: `Authorization`, VALUE: `Token
9ba45f09651c5b0c404f37a2d2572c026c14669c` + KEY: `Content Type`, VALUE: `application/json` + KEY: `Accept`, VALUE: `application/json`. These should be three new headers, and there will be two Accept headers.
- [ ] Confirm there is a Saab in this cart.
- [ ] Navigate to postman, run GET on `http://localhost:8000/lineitems/4`; in Headers, pass: KEY: `Authorization`, VALUE: `Token
9ba45f09651c5b0c404f37a2d2572c026c14669c`
- [ ] Confirm a Saab on `order_id: 2` is returned
- [ ] Run DELETE on `http://localhost:8000/lineitems/4`; in Headers, pass: KEY: `Authorization`, VALUE: `Token
9ba45f09651c5b0c404f37a2d2572c026c14669c`
- [ ] Confirm 204 status code and empty response is returned
- [ ]  Run GET on `http://localhost:8000/lineitems/4`; in Headers, pass: KEY: `Authorization`, VALUE: `Token
9ba45f09651c5b0c404f37a2d2572c026c14669c`
- [ ] Confirm 404 status code and: `{
    "message": "OrderProduct matching query does not exist."
}`
- [ ] Run GET on `http://localhost:8000/profile/cart`; in Headers, pass: KEY: `Authorization`, VALUE: `Token
9ba45f09651c5b0c404f37a2d2572c026c14669c` + KEY: `Content Type`, VALUE: `application/json` + KEY: `Accept`, VALUE: `application/json`. These should be three new headers, and there will be two Accept headers.
- [ ] Confirm there is no Saab in this cart.

## Related Issues

- Fixes #20

### CODE REVIEW

File: bangazonapi/views/lineitem.py [hunk 1]
```diff
@@ -79,8 +90,9 @@ def destroy(self, request, pk=None):
         """
         try:
             customer = Customer.objects.get(user=request.auth.user)
-            order_product = OrderProduct.objects.get(
+            line_item = OrderProduct.objects.get(
                 pk=pk, order__customer=customer)
+            line_item.delete()
```
Comment (hannahhall) [inline bangazonapi/views/lineitem.py:84]: This is the only line that needed to be added to fix this ticket. Adding extra code that doesn't have a purpose can be confusing later when working on a project

**Review by hannahhall — CHANGES_REQUESTED** [review]
Just remove the list method that you added then it'll be good

File: bangazonapi/views/lineitem.py (hunk 1 above)
Comment (gqgonzales) [reply bangazonapi/views/lineitem.py:84]: New commit pushed, list method removed

**Review by heymonicakay — APPROVED** [review]
Nice work! Approved!

**Pull request merged**